volttron/platform/install_agents.py

```diff
@@ -239,95 +202,117 @@ def install_agent_directory(opts):
-                                                         channel_name)
-            _log.debug('calling control install agent.')
-            agent_uuid = opts.connection.call_no_get('install_agent',
-                                                     filename,
-                                                     channel_name,
-                                                     vip_identity=vip_identity,
-                                                     publickey=publickey,
-                                                     secretkey=secretkey)
-
-            _log.debug('Sending wheel to control')
-            sha512 = hashlib.sha512()
-            with open(filename, 'rb') as wheel_file_data:
-                while True:
-                    # get a request
-                    with gevent.Timeout(60):
-                        request, file_offset, chunk_size = channel.recv_multipart()
-                    if request == b'checksum':
-                        channel.send(sha512.digest())
-                        break
+        opts.package = opts.install_path
+        if not os.path.exists(opts.package):
+            raise FileNotFoundError(f"Invalid file {opts.package}")
+        _send_and_intialize_agent(opts, publickey, secretkey)
+        
+    # This is where we need to exit so the script doesn't continue after installation.
+    sys.exit(0)
 
-                    assert request == b'fetch'
-
-                    # send a chunk of the file
-                    file_offset = int(file_offset)
-                    chunk_size = int(chunk_size)
-                    wheel_file_data.seek(file_offset)
-                    data = wheel_file_data.read(chunk_size)
-                    sha512.update(data)
-                    channel.send(data)
-
-            agent_uuid = agent_uuid.get(timeout=10)
-
-        except Exception as exc:
-            if opts.debug:
-                traceback.print_exc()
-            _stderr.write(
-                '{}: error: {}: {}\n'.format(opts.command, exc, filename))
-            return 10
-        else:
-            if tag:
-                opts.connection.call('tag_agent',
-                                     agent_uuid,
-                                     tag)
-        finally:
-            _log.debug('closing channel')
-            if channel:
-                channel.close(linger=0)
-                del channel
 
-    name = opts.connection.call('agent_name', agent_uuid)
-    _stdout.write('Installed {} as {} {}\n'.format(filename, agent_uuid, name))
+def _send_agent(connection, peer, path,
+                vip_identity, publickey, secretkey, force):
+    wheel = open(path, 'rb')
+    _log.debug(f"Connecting to {peer} to install {path}")
+    channel = connection.vip.channel(peer, 'agent_sender')
 
-    opts.connection.server.core.stop()
+    def send():
+        nonlocal wheel, channel
+        sha512 = hashlib.sha512()
+        try:
+            # TODO: RMQ channel???
```

Comment:
With rmq instance "vctl install examples/ListenerAgent " prints "additional arguments needed"

Reply:
yeah I need some help with the rmq side of things.